### PR TITLE
Add pep8-naming to the conventions

### PR DESCRIPTION
**Pull request by thejcannon**
This plugin only enforces rules already specified by PEP8, so it should be a slam-dunk!

### CODE REVIEW

**Review by irwand — APPROVED** [review]

**Review by DavidCurtiss — APPROVED** [review]

**Review by mshafer-NI — APPROVED** [review]

**Review by alejandro5042 — APPROVED** [review]

**Review by kcuzner — APPROVED** [review]

**Review by neilvana — APPROVED** [review]

**Review by rick-morrow — APPROVED** [review]

**Comment by thejcannon** [comment]
Nag: @alexweav @jryckman @nethrasuresh @pakdev @rtzoeller @sbethur @stick152

**Review by rtzoeller — APPROVED** [review]

**Review by alexweav — APPROVED** [review]

File: docs/Coding-Conventions.md [hunk 1]
```diff
@@ -770,6 +800,45 @@ import cheese_shop.brie
...
+```
+
+```python
+# Bad - will produce N814
+from difflib import HtmlDiff as HTML_DIFF
+```
+
+```python
+# Bad - will produce N817
+from difflib import SequenceMatcher as sm
```
Comment (pakdev) [inline docs/Coding-Conventions.md:834]: Is SM allowed for this example?

**Review by pakdev — APPROVED** [review]

File: docs/Coding-Conventions.md (hunk 1 above)
Comment (thejcannon) [reply docs/Coding-Conventions.md:834]: From local testing, looks like it. Personally, I'm not a fan of abbreviating things, but for this rule as long as you don't change the case you're OK.

Comment (mshafer-NI) [reply docs/Coding-Conventions.md:834]: should we add a "better" example?

Comment (thejcannon) [reply docs/Coding-Conventions.md:834]: Specifically N817 is "don't change case when abbreviating". What are you suggesting?

Comment (mshafer-NI) [reply docs/Coding-Conventions.md:834]: SM would be better, but not abbreviating would be best, yes??

Or do we want to just leave this for now unless we pick it up with a different linter rule?

File: docs/Coding-Conventions.md [hunk 2]
```diff
@@ -770,6 +800,45 @@ import cheese_shop.brie
...
+from difflib import HtmlDiff as htmldiff
+```
+
+```python
+# Bad - will produce N814
+from difflib import HtmlDiff as HTML_DIFF
+```
+
+```python
+# Bad - will produce N814
```
Comment (pakdev) [inline docs/Coding-Conventions.md:828]: Are these lines dupes of 823-4?

File: docs/Coding-Conventions.md [hunk 3]
```diff
@@ -770,6 +800,45 @@ import cheese_shop.brie
...
+from difflib import HtmlDiff as HTML_DIFF
+```
```
Comment (pakdev) [inline docs/Coding-Conventions.md:830]: Are these lines dupes of 822-25

**Pull request merged**